### PR TITLE
Bug 1595589 - Add all scopes required by bugbug hooks

### PR DESCRIPTION
**Pull request by marco-c**

### CODE REVIEW

File: config/projects.yml [hunk 1]
```diff
@@ -549,12 +549,20 @@ relman:
...
+      to: repo:github.com/mozilla/bugbug:*
+    # The build scopes for the tag will come from the previous rule.
     - grant: assume:project:relman:bugbug/deploy
       to: repo:github.com/mozilla/bugbug:tag:*
     - grant:
-        - "queue:route:notify.email.release-mgmt-analysis@mozilla.com.on-failed"
-        - "queue:route:notify.irc-channel.#bugbug.on-failed"
-        - "queue:route:index.project.relman.bugbug.data-pipeline-start"
+        - assume:project:relman:bugbug/build
+        - assume:project:relman:bugbug/deploy
```
Comment (marco-c) [inline config/projects.yml]: Is the "circular dependency" acceptable?

File: config/projects.yml [hunk 2]
```diff
@@ -549,12 +549,20 @@ relman:
...
+        - hooks:trigger-hook:project-relman/bugbug*
+        - queue:route:notify.email.release-mgmt-analysis@mozilla.com.on-failed
+        - queue:route:notify.irc-channel.#bugbug.on-failed
+        - queue:route:index.project.relman.bugbug.*
+        - queue:route:index.project.relman.bugbug_annotate.*
```
Comment (marco-c) [inline config/projects.yml]: Would a merge of the two, "queue:route:index.project.relman.bugbug*", work?

File: config/projects.yml [hunk 3]
```diff
@@ -549,12 +549,20 @@ relman:
         - queue:route:project.relman.bugbug.deploy_ending.*
```
Comment (marco-c) [inline config/projects.yml]: Do we need "queue:route:project.relman.bugbug.deploy_ending" and "queue:route:project.relman.bugbug.deploy_ending.*"? In bugbug I have https://github.com/mozilla/bugbug/blob/da439431d650e028c9f40d6e2d82ff3dcec7f4a9/.taskcluster.yml#L321 and https://github.com/mozilla/bugbug/blob/da439431d650e028c9f40d6e2d82ff3dcec7f4a9/infra/taskcluster-hook-pipeline-start.json#L5.

File: config/projects.yml (hunk 2 above)
Comment (djmitche) [reply config/projects.yml]: it would!

File: config/projects.yml (hunk 3 above)
Comment (djmitche) [reply config/projects.yml]: `deploy_ending*` should do, then.

File: config/projects.yml (hunk 1 above)
Comment (djmitche) [reply config/projects.yml]: If the auth service doesn't reject it, it's OK.

File: config/projects.yml (hunk 3 above)
Comment (marco-c) [reply config/projects.yml]: OK, done.

File: config/projects.yml (hunk 2 above)
Comment (marco-c) [reply config/projects.yml]: OK, done.

File: config/projects.yml (hunk 1 above)
Comment (djmitche) [reply config/projects.yml]: `taskcluster.exceptions.TaskclusterRestFailure: Roles may not contain dependency cycles, found: 'assume:hook-id:project-relman/bugbug*' -> 'assume:project:relman:bugbug/deploy' -> 'assume:hook-id:project-relman/bugbug*'`

**Comment by djmitche** [comment]
Sorry, I accidentally reverted this since I only clicked "merge" once in the UI.  I shouldn't do things while in meetings!

**Comment by djmitche** [comment]
(it's applied now)

**Pull request merged**